packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -126,6 +145,12 @@ export default function createCssVarsProvider(options) {
       }
     }, [colorScheme, attribute]);
 
+    React.useEffect(() => {
+      if (disableTransitionOnChange) {
+        disableCSSTransition();
+      }
+    }, [setMode]);
```

Comment:
Shouldn't the dependency array include colorScheme? How does `disableCSSTransition` get called if the useEffect only contain `setMode`?